src/main/java/org/wildfly/client/config/ConfigurationXMLStreamReader.java

```diff
@@ -725,13 +726,57 @@ default Expression getExpressionAttributeValue(int index, Expression.Flag... fla
         final String attributeValue = getAttributeValue(index);
         if (attributeValue == null) {
             return null;
+        } else if (attributeValue.startsWith("env.")) {
+            String envVar = System.getenv().get(attributeValue.substring(4));
+
+            if (envVar.startsWith("ENC:")) {
+                return resolveEncryptedExpression(envVar, flags);
+            } else {
+                return Expression.compile(envVar, flags);
+            }
+        } else if (attributeValue.contains("env") || attributeValue.contains("ENV")) {
+            String envVar = replaceNonAlphanumericByUnderscoresAndMakeUpperCase(attributeValue);
+
+            if (envVar.startsWith("ENC:")) {
+                return resolveEncryptedExpression(envVar, flags);
+            } else {
+                return Expression.compile(envVar, flags);
+            }
+        } else if (attributeValue.startsWith("prop.")) {
+            String propertyValue = System.getProperty(attributeValue.substring(5));
+
+            if (propertyValue.startsWith("ENC:")) {
+                return resolveEncryptedExpression(propertyValue, flags);
+            } else {
+                return Expression.compile(propertyValue, flags);
+            }
+        } else if (attributeValue.contains("prop") || attributeValue.contains("PROP")) {
+            String propertyValue = System.getProperty(replaceNonAlphanumericByUnderscoresAndMakeUpperCase(attributeValue));
+
+            if (propertyValue.startsWith("ENC:")) {
+                return resolveEncryptedExpression(propertyValue, flags);
+            } else {
+                return Expression.compile(propertyValue, flags);
+            }
+        } else if (attributeValue.contains("ENC:")) {
+            return resolveEncryptedExpression(attributeValue, flags);
         } else try {
             return Expression.compile(attributeValue, flags);
         } catch (IllegalArgumentException ex) {
             throw msg.expressionParseException(ex, getAttributeName(index), getLocation());
         }
     }
 
+    default Expression resolveEncryptedExpression(String attributeValue,  Expression.Flag... flags) throws ConfigXMLParseException {
+        Iterator resolverProviderIterator = ServiceLoader.load(ResolverProvider.class).iterator();
+        if (resolverProviderIterator.hasNext()) {
+            final ResolverProvider resolverProvider = (ResolverProvider) resolverProviderIterator.next();
```

Comment:
What is the expected behavior if there are multiple providers? First one wins?

Reply:
Currently that is what is happening. I think that is what is being done [here](https://github.com/resteasy/resteasy/blob/main/resteasy-client/src/main/java/org/jboss/resteasy/client/jaxrs/engines/ClientHttpEngineBuilder43.java#L244-L245), which is what I used as a reference. 

Could there be cases where there are multiple providers? If that from multiple places having classes that implements ResolverProvider interface? If that is the case, then we should be ok for now, since the Elytron project is the only one so far that does it. This is the implementation: https://github.com/wildfly-security/wildfly-elytron/pull/2084/files#diff-26ddef11ba5f01d385a8e42995abfc408764299ab18c6c596d8e1c82da6502a5

Reply:
Perfect! 
I will add a comment in the code to note that in case someone wants to change the policy in the future.